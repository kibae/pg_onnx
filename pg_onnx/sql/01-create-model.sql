SELECT *
FROM pg_onnx_inspect_model_bin(
        PG_READ_BINARY_FILE('../../../onnxruntime-server/test/fixture/sample/1/model.onnx')::bytea
    );


SELECT pg_onnx_create_model(
               'sample_model',
               '1',
               PG_READ_BINARY_FILE('../../../onnxruntime-server/test/fixture/sample/1/model.onnx')::bytea,
               '{}',
               'sample model'
           );

SELECT pg_onnx_create_model(
               'sample_model',
               '1',
               PG_READ_BINARY_FILE('../../../onnxruntime-server/test/fixture/sample/1/model.onnx')::bytea,
               '{}',
               'sample model'
           );

SELECT pg_onnx_create_model(
               'sample_model',
               '2',
               PG_READ_BINARY_FILE('../../../onnxruntime-server/test/fixture/sample/2/model.onnx')::bytea,
               '{}',
               'sample model'
           );

SELECT name, version, option, inputs, outputs, description
FROM ext_pg_onnx.model
ORDER BY 1, 2;

SELECT name, version, option, inputs, outputs, description
FROM pg_onnx_list_model()
ORDER BY 1, 2;
