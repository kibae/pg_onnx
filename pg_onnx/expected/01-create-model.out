SELECT *
FROM pg_onnx_inspect_model_bin(
        PG_READ_BINARY_FILE('../../../onnxruntime-server/test/fixture/sample/1/model.onnx')::bytea
    );
                               inputs                               |           outputs           
--------------------------------------------------------------------+-----------------------------
 {"x": "float32[-1,1]", "y": "float32[-1,1]", "z": "float32[-1,1]"} | {"output": "float32[-1,1]"}
(1 row)

SELECT pg_onnx_create_model(
               'sample_model',
               '1',
               PG_READ_BINARY_FILE('../../../onnxruntime-server/test/fixture/sample/1/model.onnx')::bytea,
               '{}',
               'sample model'
           );
 pg_onnx_create_model 
----------------------
 t
(1 row)

SELECT pg_onnx_create_model(
               'sample_model',
               '1',
               PG_READ_BINARY_FILE('../../../onnxruntime-server/test/fixture/sample/1/model.onnx')::bytea,
               '{}',
               'sample model'
           );
ERROR:  duplicate key value violates unique constraint "model_bin_pkey"
DETAIL:  Key (name, version)=(sample_model, 1) already exists.
CONTEXT:  SQL statement "WITH res
             AS (
            INSERT INTO ext_pg_onnx.model_bin (name, version, model)
                VALUES ($1, $2, $3)
                RETURNING model_bin.name, model_bin.version)
    SELECT CASE WHEN COUNT(*) > 0 THEN TRUE ELSE FALSE END
                    FROM res"
PL/pgSQL function pg_onnx_create_model(text,text,bytea,jsonb,text) line 6 at SQL statement
SELECT pg_onnx_create_model(
               'sample_model',
               '2',
               PG_READ_BINARY_FILE('../../../onnxruntime-server/test/fixture/sample/2/model.onnx')::bytea,
               '{}',
               'sample model'
           );
 pg_onnx_create_model 
----------------------
 t
(1 row)

SELECT name, version, option, inputs, outputs, description
FROM ext_pg_onnx.model
ORDER BY 1, 2;
     name     | version | option |                               inputs                               |           outputs           | description  
--------------+---------+--------+--------------------------------------------------------------------+-----------------------------+--------------
 sample_model | 1       | {}     | {"x": "float32[-1,1]", "y": "float32[-1,1]", "z": "float32[-1,1]"} | {"output": "float32[-1,1]"} | sample model
 sample_model | 2       | {}     | {"input_ids": "int64[-1,512]", "onnx::Equal_1": "int64[-1,512]"}   | {"output": "int64[-1]"}     | sample model
(2 rows)

SELECT name, version, option, inputs, outputs, description
FROM pg_onnx_list_model()
ORDER BY 1, 2;
     name     | version | option |                               inputs                               |           outputs           | description  
--------------+---------+--------+--------------------------------------------------------------------+-----------------------------+--------------
 sample_model | 1       | {}     | {"x": "float32[-1,1]", "y": "float32[-1,1]", "z": "float32[-1,1]"} | {"output": "float32[-1,1]"} | sample model
 sample_model | 2       | {}     | {"input_ids": "int64[-1,512]", "onnx::Equal_1": "int64[-1,512]"}   | {"output": "int64[-1]"}     | sample model
(2 rows)

